bip-musig2/reference.py

```diff
@@ -227,8 +226,6 @@ def nonce_gen(sk: bytes, aggpk: bytes, msg: bytes, extra_in: bytes) -> Tuple[byt
         raise ValueError('The optional byte array sk must have length 0 or 32.')
     if len(aggpk) not in (0, 32):
         raise ValueError('The optional byte array aggpk must have length 0 or 32.')
-    if len(msg) not in (0, 32):
-        raise ValueError('The optional byte array msg must have length 0 or 32.')
```

Comment:
One easy way to test longer messages might be to change `test_sign_and_verify_random`; the line that generates messages currently does `msg = secrets.token_bytes(32)` but any length should be OK. (Note that we'd also need to update `schnorr_verify` to remove the message length check.)

Reply:
Interesting idea, but I'm not sure if we want to change schnorr_verify at this point.